The Doom Tree

grondag opened the pull request:
See #143 for explanation.

Still a bit rough and features missing but shouldn't break things,

grondag committed:
The Doom Tree

grondag committed:
The Doom Tree

grondag committed:
The Doom Tree

grondag committed:
The Doom Tree

Draylar committed:
fully finish blood recipe type

Draylar committed:
Merge branch 'master' of https://github.com/fabric-community/spooky-time into recipetype

# Conflicts:
#	src/main/java/com/fabriccommunity/spookytime/SpookyTime.java

i509VCB committed:
All the Pumpkins Part 1.

i509VCB committed:
Merge branch 'master' of https://github.com/fabric-community/spooky-time into pr-1

# Conflicts:
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyBlocks.java
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyItems.java
#	src/main/resources/assets/spookytime/lang/en_us.json
#	src/main/resources/spookytime.mixins.json

i509VCB committed:
Add all Color textures.

i509VCB committed:
Fix mini-errors

i509VCB committed:
Merge branch 'master' of https://github.com/fabric-community/spooky-time into pr-1

# Conflicts:
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyBlocks.java
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyItems.java
#	src/main/resources/assets/spookytime/lang/en_us.json
#	src/main/resources/spookytime.mixins.json

i509VCB committed:
Add colored Pumpkins to spooky Pumpkin Feature. Add english translations for carved and jack o'lantern variants.

i509VCB committed:
Move SnowGolem things to API

i509VCB committed:
Merge branch 'master' of https://github.com/fabric-community/spooky-time into pr-1

# Conflicts:
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyBlocks.java

i509VCB committed:
Add all loot tables, add rainbow pumpkin, add pumpkin seed recipe, add item group for pumpkins.

i509VCB committed:
Snow golems created with witched pumpkins create lighting upon summon

i509VCB committed:
Merge branch 'master' of https://github.com/fabric-community/spooky-time into pr-1

# Conflicts:
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyFeatures.java
#	src/main/java/com/fabriccommunity/spookytime/world/biome/HauntedMoorBiome.java

Virtuoel committed:
Gradle & .gitignore tweaks and cleanup

Remove redundant processResources
Add MC version to version
Consistent project and refmap naming
Add GitHub releases publishing
Consistent quotes

Unignore Gradle wrapper files
Remove redundant .ini entries
Ignore .DS_STORE and similar files

i509VCB committed:
Try to make recipes work again

i509VCB committed:
Merge branch 'master' of https://github.com/fabric-community/spooky-time into pr-1

# Conflicts:
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyItems.java
#	src/main/resources/spookytime.mixins.json

i509VCB committed:
Fix imports and merge upstream

# Conflicts:
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyBlocks.java
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyFeatures.java
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyItems.java
#	src/main/java/com/fabriccommunity/spookytime/world/biome/HauntedMoorBiome.java
#	src/main/resources/spookytime.mixins.json

i509VCB committed:
Add Pumpkin Food recipes so they work with colored pumpkins

i509VCB committed:
Use Tags for pumpkins and fix formatting.

Shnupbups committed:
stuff

i509VCB committed:
Formatting II

Commit:
Darked and desaturate deceased grass, disable randoms and rotation (looks odd, we'll get that fixed up later), and make deadwood leaves fit the deadwood log better.

Commit:
Set the Environment to CLIENT on certain methods which need it.

Commit:
Clean up setting default water and stone for the spooky dimension.

Commit:
Do not register our custom ChunkGeneratorType. This fixes server-side compat without seemingly any issues. I don't think ChunkGeneratorType really needs to be synced in the first place.

Commit:
Change build.gradle to include World Traveler JiJ.

Commit:
Register the ChunkGeneratorType again.

Commit:
Fix server compat the proper way.

Shnupbups committed:
Merge pull request #141 from Virtuoel/gradle_cleanup_pr

Gradle & .gitignore tweaks and cleanup

Shnupbups reviewed (changes requested):
i don't understand most of this but i love it
I would like to ask you to remove `_1` and similar from all variable names though. Also adding localisation to en_us.json would be nice.
Is there a reason all the Doomed blocks make sand sounds?

Well done though, the Doom Tree looks insane and I love how it grows.

Shnupbups commented on src/main/java/com/fabriccommunity/spookytime/doomtree/DoomTreeClient.java:
```diff
@@ -0,0 +1,51 @@
+		InvalidateRenderStateCallback.EVENT.register(() -> {
+			DoomTree.RENDER_REFRESH_HANDLER = p -> {
+				final MinecraftClient client = MinecraftClient.getInstance();
+				System.out.println("boop");
```
also remove this println

ImAK9 replied:
Boop

grondag commented:
The doomed blocks are no longer their original material. They break instantly and crumble into doomed residue. So sand seemed like the appropriate block sound. That said, I haven’t done any sound work yet. 

Localization will be done and I will clean up the _1 names I can find. Some of them are placeholders. 

You’re welcome to let this marinate until it’s in a more complete state.

Shnupbups commented on src/main/java/com/fabriccommunity/spookytime/doomtree/DoomTree.java:
```diff
@@ -0,0 +1,102 @@
+	public static Block ALCHEMICAL_BASIN_BLOCK = SpookyBlocks.register("alchemical_basin", new AlchemicalBasinBlock(FabricBlockSettings.of(Material.METAL).breakByHand(true).strength(0.5F, 0.5F).build()));
+	
+	public static final BlockEntityType<AlchemicalBasinBlockEntity> ALCHEMICAL_BASIN =
+			Registry.register(Registry.BLOCK_ENTITY, SpookyTime.id("alchemical_basin"), BlockEntityType.Builder.create(AlchemicalBasinBlockEntity::new, DOOM_HEART_BLOCK).build(null));
```
The BE type should be applicable to the Basin block, why is it set to the Doom Heart block? This leads to the game being angry about invalid BEs

grondag replied:
'cuz it was wrong. :-). Thanks for spotting.  Can you tell I was in a hurry?

Shnupbups commented on src/main/java/com/fabriccommunity/spookytime/doomtree/DoomTree.java:
```diff
@@ -0,0 +1,102 @@
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.registry.Registry;
+
+public class DoomTree {
```
why are you doing all the registry stuff in here instead of existing registry classes like SpookyBlocks and SpookyItems?

grondag replied:
Because every single time you do a merge I have to fix up my commits.  It can be reintegrated when I'm not changing things as regularly, but IMO organizing by feature set would have been preferable to organizing by registry type when you have many people making simultaneous changes.

Shnupbups commented on src/main/java/com/fabriccommunity/spookytime/doomtree/DoomTree.java:
```diff
@@ -0,0 +1,102 @@
+	public static Block DOOMED_RESIDUE_BLOCK = SpookyBlocks.register("doomed_residue_block", new Block(FabricBlockSettings.of(Material.EARTH).breakByHand(true).sounds(BlockSoundGroup.SAND).build()));
+	public static Block WARDING_ESSENCE_BLOCK = SpookyBlocks.register("warding_essence_block", new Block(FabricBlockSettings.of(Material.EARTH).breakByHand(true).sounds(BlockSoundGroup.SAND).build()));
+	
+	public static Item DOOMED_RESIDUE_ITEM = Registry.register(Registry.ITEM, SpookyTime.id("doomed_residue"), new Item((new Item.Settings()).group(SpookyTime.GROUP)));
```
Why are you using `SpookyBlocks.register` for blocks but not using `SpookyItems.register` for items? Also we have a helper method in SpookyItems for getting a simple Item Settings.

grondag replied:
SpookyBlocks.register() is actually useful enough that I changed to make it public vs duplicate it.  All the other registration methods are protected so I didn't use 'em.

Commit:
Add basic tombstone object with texture. NOT DONE.

grondag committed:
Remove cruft, inadvertent changes

grondag committed:
Remove cruft, inadvertent changes

grondag committed:
Remove cruft, inadvertent changes

grondag committed:
Remove cruft, inadvertent changes

grondag committed:
Clean up and english localization

grondag committed:
Clean up and english localization

grondag committed:
Clean up and english localization

Commit:
Revert Deadwood Leaves texture.

Shnupbups committed:
Merge pull request #144 from Zundrel/master

Make the dimension textures spookier as well as fix up server compat.

Shnupbups committed:
cultist

model doesn't look right in game yet, very unfinished

Commit:
Working tombstone block added (although no rotation).

avayert committed:
Update README.md

Since the word `ideas` is plural, the word `them` (plural) makes more sense than `it` (singular).

vini2003 committed:
Merge pull request #147 from avayert/patch-1

Update README.md

Commit:
Fix import statements.

Commit:
retore mistakenly deleted lang-file.

Commit:
Merge branch 'master' into master

Commit:
Add descriptors to the various mixins.

Commit:
Actually remap this since it was set not to for some ungodly reason.

Commit:
Fix our refmap name.

grondag committed:
Basin fueling

grondag committed:
Basin fueling

grondag committed:
Basin fueling

grondag committed:
Basin fueling

grondag committed:
Merge branch 'master' of https://github.com/grondag/spooky-time

Draylar committed:
Update README.md

valoeghese committed:
stone circle chest, 1/3 of the time

valoeghese committed:
this should fix the chest placement

valoeghese committed:
ok this should fix it. also some tweaks

valoeghese committed:
ok it works this time I tested it

Commit:
Fixed placement according to player position.

Commit:
Merge branch 'master' of github.com:vronc/spooky-time

Commit:
Change rendering type

Shnupbups committed:
Merge pull request #149 from Zundrel/master

Add method descriptors to the various mixins.

Commit:
Remove cullface specification from model.

Juuxel committed:
Merge pull request #150 from valoeghese/dev-main

Stone circle tweaks and loot chest 1/3 of time therein

Commit:
Add bottom face to tombstone.

Commit:
Remove most mob spawns in preparation for adding our own custom mobs. Only spawn Pumpcowns in Pumpkin Patches. Re-add tinting for our custom grass, tint pumpkin patches brown.

Commit:
Darken the grass in swamps and forests. Modify Deadwood wood textures.

Commit:
Fix server crashes caused by trick or treating and the treasure chest.

Commit:
Fix whitespace.

Shnupbups committed:
lots of stuff

Shnupbups committed:
Merge pull request #151 from Zundrel/master

Dimension Improvements, Texture Changes, and Server Crash Fixes

Shnupbups committed:
Merge remote-tracking branch 'upstream/master' into features

Shnupbups committed:
cleanup

Shnupbups committed:
Merge pull request #152 from Shnupbups/features

Tons of new things

Commit:
Fix missing imports.

vini2003 committed:
Merge pull request #154 from Zundrel/master

Fix missing imports.

Commit:
Merge branch 'master' into master

Commit:
Remove whitespace

Commit:
Merge pull request #148 from vronc/master

#138 Add decorative Tombstone

Zundrel commented:
![](https://github.trello.services/images/mini-trello-icon.png) [The Doom Tree](https://trello.com/c/9RmYaYpv/3-the-doom-tree)

Draylar committed:
Merge branch 'master' of https://github.com/fabric-community/spooky-time into recipetype

# Conflicts:
#	src/main/java/com/fabriccommunity/spookytime/SpookyTime.java
#	src/main/java/com/fabriccommunity/spookytime/block/BloodBlock.java

Draylar committed:
re-add leather dying

Draylar committed:
add witch water crafting support, fix conflicts, add 2 generic recipe examples

Commit:
Merge pull request #105 from Draylar/recipetype

Blood Crafting Recipe Type

i509VCB committed:
Fix improperly named tags

i509VCB committed:
Merge branch 'master' of https://github.com/fabric-community/spooky-time into pr-1

# Conflicts:
#	src/main/java/com/fabriccommunity/spookytime/block/BloodBlock.java
#	src/main/java/com/fabriccommunity/spookytime/block/CraftingFluidBlock.java
#	src/main/java/com/fabriccommunity/spookytime/block/WitchWaterBlock.java
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyBlocks.java
#	src/main/java/com/fabriccommunity/spookytime/registry/SpookyItems.java

i509VCB committed:
Witched pumpkin works with anvil now.

vini2003 committed:
Merge pull request #123 from i509VCB/pr-1

All the Pumpkins Part 1.

grondag committed:
Start of tools and such

grondag committed:
Start of tools and such

grondag committed:
Post-merge cleanup

grondag committed:
Merge branch 'master' of https://github.com/grondag/spooky-time

grondag commented:
The real horrors are the merge conflicts.  Looks like git or GitHub shat the bed so will have to redo this PR.
